node/test/hyperbahn/hostports.js

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+'use strict';
+
+var DebugLogtron = require('debug-logtron');
```

Comment:
We need more tests.

Let's add a bunch of tests for thrift edge cases.

 - test sending a request with a malformed thrift IDL where we send an empty body. Assert BadRequest
 - test sending a request with a malformed thrift IDL where we send a body with a query without the serviceName field. Assert BadRequest
 - test sending a request with correct thrift IDL where we send an empty string service Name. Assert NotOk application error
 - test sending a request with a malformed thrift IDL which has zero exceptions defined; Send a request with empty string serviceName. Assert that the client doesn't blow up and gets back a not ok response.